crates/red_knot_python_semantic/src/types/class.rs

```diff
@@ -409,6 +424,13 @@ impl<'db> Class<'db> {
                     dynamic_type_to_intersect_with.get_or_insert(Type::from(superclass));
                 }
                 ClassBase::Class(class) => {
+                    if class.is_known(db, KnownClass::Object)
+                        // Only exclude `object` members if this is not an `object` class itself
+                        && (policy == MroAttributeLookupPolicy::NoObjectFallback && !self.is_known(db, KnownClass::Object))
```

Comment:
It seems like we don't have any tests that fail if we remove the `!self.is_known(db, KnownClass::Object)` check? (That is, tests directly looking up an attribute on `object`.)

Hmm, you do have an `object.__new__(cls)` in the tests you added; why doesn't that fail?